bindings/Fortran/CMakeLists.txt

```diff
@@ -8,6 +8,32 @@ include(FortranCInterface)
 FortranCInterface_HEADER(FC.h MACRO_NAMESPACE "FC_")
 FortranCInterface_VERIFY(CXX QUIET)
 
+# Check whether the compiler supports Fortran submodule constructs we need.
+include(CheckFortranSourceCompiles)
+CHECK_Fortran_SOURCE_COMPILES([[
+module foo
+  interface bar
+    module subroutine bar_integer(x)
+      integer, intent(in) :: x
+    end subroutine
+    module subroutine bar_real(x)
+      real, intent(in) :: x
+    end subroutine
+  end interface
+end module
+submodule ( foo ) sub
+contains
+  module subroutine bar_integer(x)
+    integer, intent(in) :: x
+  end subroutine
+  module subroutine bar_real(x)
+    real, intent(in) :: x
+  end subroutine
+end submodule
+program main
+end program
+]] ADIOS2_HAVE_FORTRAN_SUBMODULES SRC_EXT F90)
+
```

Comment:
I'd like this to be factored out into `cmake/ADIOSFunctions.cmake`.  That's essentially where we keep our "library" of custom cmake functions used for the build.